diff --git a/node_modules/@ckeditor/ckeditor5-utils/src/dom/createelement.js b/node_modules/@ckeditor/ckeditor5-utils/src/dom/createelement.js
index f0e5307..2566607 100644
--- a/node_modules/@ckeditor/ckeditor5-utils/src/dom/createelement.js
+++ b/node_modules/@ckeditor/ckeditor5-utils/src/dom/createelement.js
@@ -27,7 +27,8 @@ import { isString } from 'lodash-es';
  */
 export default function createElement( doc, name, attributes = {}, children = [] ) {
 	const namespace = attributes && attributes.xmlns;
-	const element = namespace ? doc.createElementNS( namespace, name ) : doc.createElement( name );
+	const options = attributes && attributes.is ? { is: attributes.is } : undefined;
+	const element = namespace ? doc.createElementNS( namespace, name, options ) : doc.createElement( name, options );
 
 	for ( const key in attributes ) {
 		element.setAttribute( key, attributes[ key ] );
