diff --git a/node_modules/@ckeditor/ckeditor5-ui/src/button/buttonview.js b/node_modules/@ckeditor/ckeditor5-ui/src/button/buttonview.js
index e08ee75..5ad41b5 100644
--- a/node_modules/@ckeditor/ckeditor5-ui/src/button/buttonview.js
+++ b/node_modules/@ckeditor/ckeditor5-ui/src/button/buttonview.js
@@ -123,6 +123,9 @@ export default class ButtonView extends View {
             },
             children: this.children,
             on: {
+                mousedown: bind.to(evt => {
+                    evt.preventDefault();
+                }),
                 click: bind.to(evt => {
                     // We can't make the button disabled using the disabled attribute, because it won't be focusable.
                     // Though, shouldn't this condition be moved to the button controller?
