diff --git a/node_modules/@ckeditor/ckeditor5-ui/src/template.js b/node_modules/@ckeditor/ckeditor5-ui/src/template.js
index 74e566d..783981e 100644
--- a/node_modules/@ckeditor/ckeditor5-ui/src/template.js
+++ b/node_modules/@ckeditor/ckeditor5-ui/src/template.js
@@ -437,7 +437,8 @@ export default class Template {
 		let node = data.node;
 
 		if ( !node ) {
-			node = data.node = document.createElementNS( this.ns || xhtmlNs, this.tag );
+			const options = this.attributes && this.attributes.is ? { is: this.attributes.is } : undefined;
+			node = data.node = document.createElementNS( this.ns || xhtmlNs, this.tag, options );
 		}
 
 		this._renderAttributes( data );
