diff --git a/node_modules/@ckeditor/ckeditor5-ui/src/template.js b/node_modules/@ckeditor/ckeditor5-ui/src/template.js
index e5a5077..b69b92e 100644
--- a/node_modules/@ckeditor/ckeditor5-ui/src/template.js
+++ b/node_modules/@ckeditor/ckeditor5-ui/src/template.js
@@ -397,7 +397,8 @@ export default class Template extends EmitterMixin() {
     _renderElement(data) {
         let node = data.node;
         if (!node) {
-            node = data.node = document.createElementNS(this.ns || xhtmlNs, this.tag);
+            const options = this.attributes && this.attributes.is ? { is: this.attributes.is } : undefined;
+            node = data.node = document.createElementNS(this.ns || xhtmlNs, this.tag, options);
         }
         this._renderAttributes(data);
         this._renderElementChildren(data);
