diff --git a/node_modules/@vue/runtime-dom/dist/runtime-dom.esm-bundler.js b/node_modules/@vue/runtime-dom/dist/runtime-dom.esm-bundler.js
index 96e0e0c..2dad703 100644
--- a/node_modules/@vue/runtime-dom/dist/runtime-dom.esm-bundler.js
+++ b/node_modules/@vue/runtime-dom/dist/runtime-dom.esm-bundler.js
@@ -278,7 +278,7 @@ prevChildren, parentComponent, parentSuspense, unmountChildren) {
 
 // Async edge case fix requires storing an event listener's attach timestamp.
 let _getNow = Date.now;
-let skipTimestampCheck = false;
+let skipTimestampCheck = true;
 if (typeof window !== 'undefined') {
     // Determine what event timestamp the browser is using. Annoyingly, the
     // timestamp can either be hi-res (relative to page load) or low-res
@@ -293,7 +293,7 @@ if (typeof window !== 'undefined') {
     // #3485: Firefox <= 53 has incorrect Event.timeStamp implementation
     // and does not fire microtasks in between event propagation, so safe to exclude.
     const ffMatch = navigator.userAgent.match(/firefox\/(\d+)/i);
-    skipTimestampCheck = !!(ffMatch && Number(ffMatch[1]) <= 53);
+    //skipTimestampCheck = !!(ffMatch && Number(ffMatch[1]) <= 53);
 }
 // To avoid the overhead of repeatedly calling performance.now(), we cache
 // and use the same timestamp for all event listeners attached in the same tick.
