diff --git a/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js b/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js
index eb15db4..5c0b300 100644
--- a/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js
+++ b/node_modules/@vue/runtime-core/dist/runtime-core.cjs.js
@@ -4280,7 +4280,8 @@ function createAppContext() {
             optionMergeStrategies: {},
             errorHandler: undefined,
             warnHandler: undefined,
-            compilerOptions: {}
+            compilerOptions: {},
+            skipEventsTimestampCheck: false
         },
         mixins: [],
         components: {},
diff --git a/node_modules/@vue/runtime-core/dist/runtime-core.cjs.prod.js b/node_modules/@vue/runtime-core/dist/runtime-core.cjs.prod.js
index 7ffb2f9..b85039b 100644
--- a/node_modules/@vue/runtime-core/dist/runtime-core.cjs.prod.js
+++ b/node_modules/@vue/runtime-core/dist/runtime-core.cjs.prod.js
@@ -3368,7 +3368,8 @@ function createAppContext() {
             optionMergeStrategies: {},
             errorHandler: undefined,
             warnHandler: undefined,
-            compilerOptions: {}
+            compilerOptions: {},
+            skipEventsTimestampCheck: false
         },
         mixins: [],
         components: {},
diff --git a/node_modules/@vue/runtime-core/dist/runtime-core.esm-bundler.js b/node_modules/@vue/runtime-core/dist/runtime-core.esm-bundler.js
index 543e24b..01cd8c0 100644
--- a/node_modules/@vue/runtime-core/dist/runtime-core.esm-bundler.js
+++ b/node_modules/@vue/runtime-core/dist/runtime-core.esm-bundler.js
@@ -4300,7 +4300,8 @@ function createAppContext() {
             optionMergeStrategies: {},
             errorHandler: undefined,
             warnHandler: undefined,
-            compilerOptions: {}
+            compilerOptions: {},
+            skipEventsTimestampCheck: false
         },
         mixins: [],
         components: {},
